common/djangoapps/student/tests/test_email.py

```diff
@@ -132,6 +134,23 @@ def _assert_activation_email(self, subject, body_fragments):
         for fragment in body_fragments:
             self.assertIn(fragment, msg.body)
 
+    @mock.patch('student.tasks.log')
+    def test_send_email_to_inactive_user(self, mock_log):
+        """
+        To verify that activation email has been sent to
+        an un-activated user(logged-in via social-auth)
+        """
+        inactive_user = UserFactory(is_active=False)
+        Registration().register(inactive_user)
+        request = RequestFactory().get(settings.SOCIAL_AUTH_INACTIVE_USER_URL)
+        request.user = inactive_user
+        with patch('edxmako.request_context.get_current_request'):
```

Comment:
Actually, we need to patch it because request context is required in the mako template. 
https://github.com/edx/edx-platform/blob/master/common/djangoapps/edxmako/request_context.py#L41-L51